fivetran_provider/sensors/fivetran.py

```diff
@@ -34,7 +34,7 @@ def __init__(
         self.connector_id = connector_id
         self.poke_interval = poke_interval
         self.hook = FivetranHook(self.fivetran_conn_id)
-        self.previous_completed_at = self.hook.get_last_sync(self.connector_id)
 
     def poke(self, context):
-       return self.hook.get_sync_status(self.connector_id, self.previous_completed_at)
+        self.previous_completed_at = self.hook.get_last_sync(self.connector_id)
```

Comment:
If `poke` is called every interval, this Sensor will NEVER exit, because you are setting the previous_completed_at on every refresh, and that is used to tell when to stop refreshing. So it is either going to need to be passed into the Sensor (complicating things for the user) or...?

Reply:
OK I have a solution. Kind of a no-brainer actually. 
```
if self.previous_completed_at is None:
   self.previous_completed_at = self.hook.get_last_sync(self.connector_id)
```

and then set `self.previous_completed_at = None` in the constructor.

Reply:
We can do the same thing with the Hook, actually, if we want to skip instantiating it every `poke`.

Reply:
I am going to test out this init method:
```
    def __init__(
        self,
        fivetran_conn_id: str = 'fivetran',
        poke_interval: int = 60,
        connector_id=None,
        **kwargs: Any
    ) -> None:
        super().__init__(**kwargs)
        self.fivetran_conn_id = fivetran_conn_id
        self.connector_id = connector_id
        self.poke_interval = poke_interval
        self.hook = None
        self.previous_completed_at = None

    def poke(self, context):
        if self.hook is None:
            self.hook = FivetranHook(self.fivetran_conn_id)
        if self.previous_completed_at is None:
            self.previous_completed_at = self.hook.get_last_sync(self.connector_id)
        return self.hook.get_sync_status(self.connector_id, self.previous_completed_at)
```

Reply:
could this functionality and checking be pushed from sensor.poke into hook.get_sync_status, keeping the sensor super clear for users?

Reply:
Works like a dream. This should also fix the weird error that happens on first load of these DAGs.

Reply:
https://github.com/fivetran/airflow-provider-fivetran/pull/30

Reply:
> could this functionality and checking be pushed from sensor.poke into hook.get_sync_status, keeping the sensor super clear for users?

If you are very careful about using the same FivetranHook instance every `poke`, then yes. Right now I somewhat prefer this lives in the Sensor, since that keeps the Hook more about just making API calls, but I could be convinced otherwise

Reply:
The sensor's `_do_api_call` method can be improved significantly. It currently uses a while loop to wait for a success criteria after iteratively executing code. The Airflow scheduler already takes care of this functionality. 

See how the [HttpHook's `run` method](https://github.com/apache/airflow/blob/master/airflow/providers/http/hooks/http.py#L96) (analogous to Fivetran's `_do_api_call`) returns the response of `self.run_and_check()`, which is either the API response or an Exception. This is then used as the sensor's success criteria.

Reply:
`_do_api_call` is looping to ensure it retries on simple API failures, like a transitory 500 response, without failing out of the entire operation. It is unrelated to the behavior of the sensor. There may be a better way to do this with fancy requests add-on libraries, but the basic functionality would be the same.

Reply:
So for instance if you are in the middle of a five hour sync, and Fivetran is updating the API service, you might get 500s for a few seconds (I have personally experienced this). We don't want to claim the sync has failed due to that issue.

Reply:
I think this would make sense to open as an issue, so we can dig in as a separate concern to this PR.